src/chart_process/bms_processor.rs

```diff
@@ -787,3 +564,433 @@ enum FlowEvent {
+        }
+
+        // Stop events
+        for stop in bms.stop.stops.values() {
+            let y = BmsProcessor::y_of_time_static(bms, stop.time, &bms.speed.speed_factor_changes);
+            let event = ChartEvent::Stop {
+                duration: stop.duration.clone(),
+            };
+
+            let y_coord = YCoordinate::from(y);
+            let evp = ChartEventWithPosition::new(
+                id_gen.next_id(),
+                y_coord.clone(),
+                event,
+                Duration::from_secs(0),
+            );
+            events_map.entry(y_coord).or_default().push(evp);
+        }
+
+        // BGA change events
+        for bga_obj in bms.bmp.bga_changes.values() {
+            let y =
+                BmsProcessor::y_of_time_static(bms, bga_obj.time, &bms.speed.speed_factor_changes);
+            let bmp_index = bga_obj.id.as_u16() as usize;
+            let event = ChartEvent::BgaChange {
+                layer: bga_obj.layer,
+                bmp_id: Some(BmpId::from(bmp_index)),
+            };
+
+            let y_coord = YCoordinate::from(y);
+            let evp = ChartEventWithPosition::new(
+                id_gen.next_id(),
+                y_coord.clone(),
+                event,
+                Duration::from_secs(0),
+            );
+            events_map.entry(y_coord).or_default().push(evp);
+        }
+
+        // BGA opacity change events (requires minor-command feature)
+
+        for (layer, opacity_changes) in &bms.bmp.bga_opacity_changes {
+            for opacity_obj in opacity_changes.values() {
+                let y = BmsProcessor::y_of_time_static(
+                    bms,
+                    opacity_obj.time,
+                    &bms.speed.speed_factor_changes,
+                );
+                let event = ChartEvent::BgaOpacityChange {
+                    layer: *layer,
+                    opacity: opacity_obj.opacity,
+                };
+
+                let y_coord = YCoordinate::from(y);
+                let evp = ChartEventWithPosition::new(
+                    id_gen.next_id(),
+                    y_coord.clone(),
+                    event,
+                    Duration::from_secs(0),
+                );
+                events_map.entry(y_coord).or_default().push(evp);
+            }
+        }
+
+        // BGA ARGB color change events (requires minor-command feature)
+
+        for (layer, argb_changes) in &bms.bmp.bga_argb_changes {
+            for argb_obj in argb_changes.values() {
+                let y = BmsProcessor::y_of_time_static(
+                    bms,
+                    argb_obj.time,
+                    &bms.speed.speed_factor_changes,
+                );
+                let argb = ((argb_obj.argb.alpha as u32) << 24)
+                    | ((argb_obj.argb.red as u32) << 16)
+                    | ((argb_obj.argb.green as u32) << 8)
+                    | (argb_obj.argb.blue as u32);
```

Comment:
This conversion should be extracted as a method of `Argb` into `bms::command::graphics` module.